tools/crm_resource_runtime.c

```diff
@@ -1704,6 +1706,40 @@ cli_resource_execute_from_params(pcmk__output_t *out, const char *rsc_name,
         crm_exit(CRM_EX_UNIMPLEMENT_FEATURE);
     }
 
+    if (pcmk__str_eq(rsc_action, "validate", pcmk__str_casei)) {
+        action = "validate-all";
+
+    } else if (pcmk__str_eq(rsc_action, "force-check", pcmk__str_casei)) {
+        action = "monitor";
+
+    } else if (pcmk__str_eq(rsc_action, "force-stop", pcmk__str_casei)) {
+        action = rsc_action+6;
+
+    } else if (pcmk__strcase_any_of(rsc_action, "force-start", "force-demote",
+                                    "force-promote", NULL)) {
+        action = rsc_action+6;
+
+        if(pe_rsc_is_clone(rsc)) {
+            GList *nodes = cli_resource_search(rsc, rsc_name, data_set);
+            if(nodes != NULL && force == FALSE) {
+                out->err(out, "It is not safe to %s %s here: the cluster claims it is already active",
+                         action, rsc->id);
+                out->err(out, "Try setting target-role=Stopped first or specifying "
+                         "the force option");
+                return CRM_EX_UNSAFE;
+            }
+
+            g_list_free_full(nodes, free);
+        }
```

Comment:
This clone test should stay in cli_resource_execute() since rsc can be NULL in this one